frontend/src/hooks/useBreadcrumb.ts

```diff
@@ -18,16 +24,39 @@ export function useBreadcrumbs() {
       '/profile': 'Dein Profil',
     };
 
-    return matches
-      .filter(({ pathname }) => pathname !== '/')
-      .map(({ pathname }) => {
-        const title = pathNameMap[pathname] || 'Kein Titel vorhanden';
-        return {
-          title,
-          path: pathname,
-        };
-      });
+    const result: Breadcrumb[] = [];
+
+    matches.forEach(({ pathname }) => {
+      if (pathname !== '/') {
+        if (pathname.startsWith('/treecluster/') && pathname !== '/treecluster/') {
+          const id = pathname.split('/')[2];
+          const treeclusterId = parseInt(id);
+          const treecluster = treeclusterDemoData().find(tc => tc.id === treeclusterId);
+          const title = treecluster ? treecluster.name : 'Kein Titel vorhanden';
+
+          // Add the 'Treecluster' segment
+          result.push({
```

Comment:
I thought about that too. But we probably dont want to have the id in the breadcrumbs. We want the name of the group, sensor etc. So if its about the treeclusters I need to specifically search for it in the treecluster data. I am not sure how to do that more generic.
I thought about something like:

if pathname contains treecluster
  Look for treeclustername with given id in treeclusterData
else if pathname contains sensor
  Look for sensorname with given id in SensorData
etc

I dont know how to change that because I need to address the different datafiles/databank tables. Maybe I could simplify the current code a bit. But I cant think of a different solution. Or are there ways in react/typescript I dont know about right now?